storageservice/modules/files/file_client.bal

```diff
@@ -430,34 +425,45 @@ public client class FileClient {
-            return true;
-        } else {
+        if (response.statusCode != http:STATUS_ACCEPTED) {
             fail error(check getErrorMessage(response));
         }
     }
 
-    # Upload a file directly into the fileshare.
+    # Upload a file directly to the fileshare.
     # 
     # + fileShareName - Name of the fileShare
     # + localFilePath - The path of the file to be uploaded
     # + azureFileName - The name of the file in Azure
     # + azureDirectoryPath - Directory path in Azure
     # + return - If success, returns true.  Else returns error
-    remote function directUpload(@display {label: "File share name"} string fileShareName, 
-                                 @display {label: "Local file path"} string localFilePath, 
-                                 @display {label: "File name in Azure"} string azureFileName, 
-                                 @display {label: "Azure directory path"} string? azureDirectoryPath = ()) 
-                                 returns @tainted @display {label: "File uploaded"} boolean|error {
+    isolated remote function directUpload(@display {label: "File Share Name"} string fileShareName, 
+                                 @display {label: "Local File Path"} string localFilePath, 
+                                 @display {label: "Azure File Name"} string azureFileName, 
+                                 @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
+                                 returns @tainted @display {label: "Response"} error? {
         file:MetaData fileMetaData = check file:getMetaData(localFilePath);
         int fileSizeInByte = fileMetaData.size;
-        var createFileResponse = self->createFile(fileShareName, azureFileName, fileSizeInByte, azureDirectoryPath);
-        if (createFileResponse === true) {
-            var uploadResult = putRangeInternal(self.httpClient, fileShareName, localFilePath, azureFileName, 
-                self.azureConfig, fileSizeInByte, azureDirectoryPath);
-            return uploadResult;
-        } else {
-            return createFileResponse;
-        }
+        check self->createFile(fileShareName, azureFileName, fileSizeInByte, azureDirectoryPath);
+        check putRangeInternal(self.httpClient, fileShareName, localFilePath, azureFileName, 
+                            self.azureConfig, fileSizeInByte, azureDirectoryPath);
+    }
+
+    # Upload a byte array directly to the fileshare.
+    # 
+    # + fileShareName - Name of the fileShare
+    # + fileContent - File content as a byte array
+    # + azureFileName - Name of the file in Azure
+    # + azureDirectoryPath - Directory path in Azure
+    # + return - If success, returns true.  Else returns error
+    isolated remote function directUploadFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
+                                 @display {label: "File Content Array"} byte[] fileContent, 
```

Comment:
`File Content (Byte Array)` - however, not useful for Choreo at all.

Reply:
Updated the Display annotations. It is useful to the user when needs to upload files that get as byte array such as Gmail attachments etc.

Reply:
+1. Yes, we can get it as an expression variable.